purchase_order_type/models/purchase_order.py

```diff
@@ -19,15 +19,35 @@
 #
 #
 
-from openerp import models, fields
+from openerp import models, fields, api
 from openerp.addons.purchase.purchase import purchase_order
 
 
 class PurchaseOrder(models.Model):
     _inherit = 'purchase.order'
 
+    def _get_order_type(self):
+        return self.env['purchase.order.type'].search([])[:1]
+
     order_type = fields.Many2one(comodel_name='purchase.order.type',
                                  readonly=False,
                                  states=purchase_order.READONLY_STATES,
                                  string='Type',
-                                 ondelete='restrict')
+                                 ondelete='restrict',
+                                 default=_get_order_type)
+
+    @api.multi
+    def onchange_partner_id(self, part):
+        res = super(PurchaseOrder, self).onchange_partner_id(part)
+        if part:
+            partner = self.env['res.partner'].browse(part)
+            res['value'].update({
+                'order_type': partner.purchase_type.id
```

Comment:
You shouldn't change to the first purchase type if the partner doesn't have any purchase type. Just let the previous (don't make any change).